week_1/Array/product_of_array_except_self/product_of_array_except_self.js

```diff
@@ -0,0 +1,34 @@
+// Given an integer array nums, return an array answer such that answer[i] is equal to the product of all the elements of nums except nums[i].
+
+// The algorithm must run in O(n) time and can't use the division operator.
+/**
+ * @param {number[]} an array of numbers
+ * @return {number[]} an array of products
+ */
+
+const productExceptSelf = (nums) => {
+	const fromLeft = [];
+	let leftMultiplier = 1;
+
+	// This loop creates an array where each index is equal to the product of all nums to the left of nums[i]
+	for (let i = 0; i < nums.length; i++) {
+		fromLeft[i] = leftMultiplier;
+		leftMultiplier *= nums[i];
+	}
+
+	const fromRight = [];
+	let rightMultiplier = 1;
+
+	const products = [];
+	// This loop creates an array where each index is equal to the product of all nums to the right of nums[i]
+	// We then multiply that index by the same index of fromLeft in order to get the product of all nums to the right and left of nums[i]
```

Comment:
A good rule of thumb for code comments is to be as concise as possible. A future version of you, or a coworker, will just want to know what's happening as quickly as possible. Being concise, almost caveman like, helps with this a lot. 

I'd also move the definition of "products" below this comment make it easier to refer to.

Something like this:
```suggestion
	// Multiple all numbers to the left and right of a given index, then store that value in a new array
```

It's also good to avoid referencing specific variable names in comments, unless it's a JSDoc comment, because those can easily be renamed without the comment being updated. The comment will still be technically accurate, but harder to understand due to referring to variables that no longer exist.